pkg/controller/vsphere/session/session_test.go

```diff
@@ -277,14 +277,16 @@ func TestClientTimeout(t *testing.T) {
 
 		model.DelayConfig = simulator.DelayConfig{
 			MethodDelay: map[string]int{
-				"RetrieveProperties": int(18 * time.Second.Milliseconds()),
+				"RetrieveProperties": int(5000 * time.Second.Milliseconds()),
 			},
 		}
 		simulatorVM := simulator.Map.Any("VirtualMachine").(*simulator.VirtualMachine)
 
-		_, err := session.findVMByName(context.TODO(), simulatorVM.Config.Name)
-		g.Expect(err.Error()).Should(ContainSubstring("unable to find template by name"))
-		g.Expect(err.Error()).Should(ContainSubstring("context deadline exceeded (Client.Timeout exceeded while awaiting headers)"))
+		ctx, cancel := context.WithTimeout(context.TODO(), time.Duration(time.Millisecond*1))
```

Comment:
So even with the extended delay in `RetrieveProperties` this is still needed? Curious

Reply:
Yep - it's weird. 

If i set the context to have a sensible timeout, e.g 200ms, when I run a `make unit`: 

```
=== RUN   TestClientTimeout/Globally_laggy_vcenter
I0604 17:02:55.660829   44336 session.go:91] No existing vCenter session found, creating new session
--- FAIL: TestClientTimeout (6.03s)
    --- PASS: TestClientTimeout/Global_delay_which_not_exceeds_the_timeout (5.31s)
    --- FAIL: TestClientTimeout/laggy_method (0.29s)
    --- PASS: TestClientTimeout/Globally_laggy_vcenter (0.43s)
```

But then with the exact same code running in a debugger the laggy_method test passes, and the error is set: `"unable to find template by name \"DC0_C0_RP0_VM1\": Post \"https://127.0.0.1:62329/sdk\": context deadline exceeded"`

Reply:
Stepping through the code with a debugger, we also do a wait of `5000`. I wonder if something is modifying the context when it's running from the make target.